### ISSUE TITLE
Wrong E0428 is shown when a macros has the same name as a type/module 

### ISSUE THREAD

**Issue by kumbayo** [intellij-rust/intellij-rust#1624]
When a macro and a module or a macro and a struct have the same name, a wrong error is shown
`A type named xxx has already been defined in this module [E0428]`

```rust
fn main() {
    println!("Hello, world!");
}

macro_rules! example {
    () => ()
}

//struct example { } // <-Error here when uncommented

mod example { } // <-Error here
```
![intellij_e0428](https://user-images.githubusercontent.com/1041267/29252880-b41d41f0-806f-11e7-9b7f-5bc174890638.png)

**Comment by kumbayo** [comment]
I am working on this one.
